bench/tiered/push_pull.c

```diff
@@ -135,35 +140,108 @@ difftime_sec(struct timeval t0, struct timeval t1)
  *     This function runs the test for configured number of times to compute the average time taken.
  */
 static void
-run_test_clean(const char *suffix, uint64_t num_records, bool flush)
+run_test_clean(const char *suffix, uint32_t num_records, bool flush)
 {
     char home_full[HOME_BUF_SIZE];
-    double avg_time, avg_throughput;
-    int64_t avg_file_size;
+    double avg_wtime, avg_rtime, avg_wthroughput, avg_rthroughput;
+    uint64_t avg_file_size;
     int counter;
 
     avg_file_size = 0;
-    avg_time = avg_throughput = 0;
+    avg_wtime = avg_rtime = avg_rthroughput = avg_wthroughput = 0;
 
     for (counter = 0; counter < MAX_RUN; ++counter) {
         testutil_check(__wt_snprintf(
           home_full, HOME_BUF_SIZE, "%s_%s_%d_%d", opts->home, suffix, flush, counter));
         run_test(home_full, num_records, flush, counter);
+    }
 
-        /* Cleanup */
-        if (!opts->preserve)
-            testutil_clean_work_dir(home_full);
+    /* Cleanup */
+    if (!opts->preserve) {
+        testutil_check(__wt_snprintf(home_full, HOME_BUF_SIZE, "%s_%s*", opts->home, suffix));
+        testutil_clean_work_dir(home_full);
     }
 
+    /* Compute the average */
     for (counter = 0; counter < MAX_RUN; ++counter) {
-        avg_time += avg_time_array[counter];
-        avg_throughput += avg_throughput_array[counter];
+        avg_wtime += avg_wtime_arr[counter];
+        avg_rtime += avg_rtime_arr[counter];
+        avg_wthroughput += avg_wthroughput_arr[counter];
+        avg_rthroughput += avg_rthroughput_arr[counter];
```

Comment:
Taking an average of throughputs (like taking the average of speeds) is not straightforward.  Suppose you have 2 runs, one is 10 megabytes in 1 second, another is 10 megabytes in 10 seconds.  Throughput for first run is 10M/sec, second run is 1M/sec and average is 5.5M/sec.  However, in total, you've transmitted 20 megabytes in 11 seconds, which is ~ 1.8 M/sec.  Which is the average?

Actually listing either number 5.5M or 1.8M alone doesn't tell the whole story.  The story here being a HUGE variation in throughput -- that definitely needs to stand out somehow. You either need an average with a standard deviation, or list all of the throughputs individually.  I vote for standard deviation, it's a bit harder, but it's a single compact number.  It can be reported as “W_Tput- 5.5MB/sec (SD = 4.7)”. The throughput "average" can be the number you are currently calculating as the average, but you'll need to figure out how to do SD.

Reply:
Thanks, @ddanderson for your suggestion to add SD. I have figured out to calculate the standard deviation for throughputs. Do we need to add SD for read/write time as well? Because in some runs I see some difference in the write time.

Reply:
Yes, I think that would be helpful to have SD for read/write time.  It's a powerful metric.

Reply:
I updated the code and added standard deviation to all the average numbers.
Now the output looks like below.

`
Running test command: ./test_push_pull -PT 
The below benchmarks are average of 5 runs
########################################################################################
                        Checkpoint is done with flush_tier disabled
########################################################################################
Records in Bytes- 122880 (~100KB), W_Time- 0.007 secs (SD- 0.000), W_Tput- 15.888 MB/sec (SD- 0.791), R_Time- 0.001 secs (SD- 0.000), R_Tput- 211.722 MB/sec
Records in Bytes- 1048576 (~1MB), W_Time- 0.022 secs (SD- 0.001), W_Tput- 44.610 MB/sec (SD- 1.349), R_Time- 0.005 secs (SD- 0.000), R_Tput- 182.029 MB/sec
Records in Bytes- 10395648 (~10MB), W_Time- 0.173 secs (SD- 0.001), W_Tput- 57.349 MB/sec (SD- 0.385), R_Time- 0.057 secs (SD- 0.000), R_Tput- 175.294 MB/sec
Records in Bytes- 104341504 (~100MB), W_Time- 1.622 secs (SD- 0.016), W_Tput- 61.362 MB/sec (SD- 0.612), R_Time- 0.576 secs (SD- 0.003), R_Tput- 172.790 MB/sec
########################################################################################
                        Checkpoint is done with flush_tier enabled
########################################################################################
Records in Bytes- 126976 (~100KB), W_Time- 0.025 secs (SD- 0.001), W_Tput- 4.847 MB/sec (SD- 0.225), R_Time- 0.001 secs (SD- 0.000), R_Tput- 219.658 MB/sec
Records in Bytes- 1052672 (~1MB), W_Time- 0.040 secs (SD- 0.001), W_Tput- 25.075 MB/sec (SD- 0.790), R_Time- 0.006 secs (SD- 0.000), R_Tput- 181.303 MB/sec
Records in Bytes- 10399744 (~10MB), W_Time- 0.190 secs (SD- 0.001), W_Tput- 52.108 MB/sec (SD- 0.256), R_Time- 0.057 secs (SD- 0.000), R_Tput- 174.958 MB/sec
Records in Bytes- 104345600 (~100MB), W_Time- 1.670 secs (SD- 0.005), W_Tput- 59.581 MB/sec (SD- 0.182), R_Time- 0.573 secs (SD- 0.001), R_Tput- 173.808 MB/sec
ubuntu@ip-10-122-7-133:/data/wiredtiger/cbuild/bench/tiered$ 

`

Reply:
What you currently have:
```
Records in Bytes- 122880 (~100KB), W_Time- 0.007 secs (SD- 0.000), W_Tput- 15.888 MB/sec (SD- 0.791), R_Time- 0.001 secs (SD- 0.000), R_Tput- 211.722 MB/sec
```
I wonder if you could change a few things, I think it would make it easier to understand:
```
Bytes transferred: 122880 (~100KB), W_Time: 0.007 secs (SD 0.000), W_Tput: 15.888 MB/sec (SD 0.791), R_Time: 0.001 secs (SD 0.000), R_Tput: 211.722 MB/sec (SD X.XXX)
```
That is Bytes transferred seems clearer to me. The use of '-' makes me think of negative numbers, but `(SD x.xxx)` without any colon or other prefix is pretty standard notation. Finally, the standard deviation is missing for read throughput?